exercises/firmware_auditing/README.md

```diff
@@ -0,0 +1,318 @@
+To do this, we perform a list comprehension, unifying with a wildcard variable to iterate over and filter all imported capabilities of all compartments in the firmware.
+We then use the built-in`count` function to ensure that the resulting array is empty.
+
+Skipping to the end of the file, we can then create a simple Boolean `valid` rule which corresponds to whether `no_sealed_capabilities_exist` is defined or not.
+To convert the undefined value to a Boolean, we use the `default` keyword, which lets us provide a `false` asignment as a fall-through if no other rule definitions match.
+
+We can run this policy on our example firmware using the following command:
+
+```sh
+cheriot-audit --board=cheriot-rtos/sdk/boards/sonata.json \
+              --firmware-report=build/cheriot/cheriot/release/firmware_auditing_part_1.json \
+              --module=exercises/firmware_auditing/part_1/no_sealed_capabilities.rego \
+              --query='data.no_seal.valid'
+```
+
+Because `sealed_capability.cc` currently doesn't use any sealed capabilities, this policy should evalute to `true`.
+
+Now, navigate to [`sealed_capability.cc`][] and comment out or remove the line `uint32_t arr[ArrSize];`.
+Then, uncomment or add the line `uint32_t *arr = new uint32_t[ArrSize];`.
+This will change the array `arr` from being allocated on the stack to the heap.
+Rebuilding the exercises (using `xmake -P exercises`) and running the same command again should now cause the policy to evaluate to `false`.
+
+[`sealed_capability.cc`]: ../../exercises/firmware_auditing/part_1/sealed_capability.cc
+
+However, it may not immediately be clear why the policy failed.
+When designing such a policy, you can see that it may be helpful to have a rule that lets us inspect the details of any sealed capabilities.
+We do this with our final `sealed_capability_info` rule, which constructs ojects storing the contents, key, and compartment of each sealed capability.
+
+> ℹ️ Note the use of unification here.
+> We iterate over `input.compartments` with the non-defined variable `owner`, and then map the value that is bound this variable to the `owner` field of our new object.
+
+Now, run the `cheriot-audit` command again, but replace `data.no_seal.valid` with `data.no_seal.sealed_capability_info`.
+You should see an output that looks something like this:
+
+```json
+[{"compartment":"alloc", "contents":"00100000 00000000 00000000 00000000 00000000 00000000",
+  "key":"MallocKey", "owner":"sealed_capability"}]
+```
+
+This tells us where the sealed capabilty in our firmware originates - a static sealed object owned by our `sealed_capability` compartment, which is used by the RTOS' allocator for authorising memory allocation.
+Try experimenting by adding more functionality to this toy example!
+For example, try adding some functionality that might cause the RTOS' scheduler's `InterruptKey` static sealed object to be added.
```

Comment:
I love the go play parts here! However, as a newbie, I'd have no idea how to do this.

Reply:
That makes sense - instead I'll suggest they try adding their own sealed capabilities (for people who already know how to do that), or change the policy itself to investigate what other types of e.g. capabilities they might filter and audit.